sdk/communication/azure-communication-chat/tests/_shared/test_communication_identity_serializer.py

```diff
@@ -90,42 +106,60 @@ def test_serialize_phone_number(self):
 
     def test_deserialize_phone_number(self):
         phone_number_identifier_actual = CommunicationUserIdentifierSerializer.deserialize(
             CommunicationIdentifierModel(
+                id="someid",
                 kind=CommunicationIdentifierKind.PhoneNumber,
                 phone_number="phonenumber"
             )
         )
 
-        phone_number_identifier_expected = PhoneNumberIdentifier("phonenumber")
+        phone_number_identifier_expected = PhoneNumberIdentifier("phonenumber", identifier="someid")
 
         assert isinstance(phone_number_identifier_actual, PhoneNumberIdentifier)
         assert phone_number_identifier_actual.phone_number == phone_number_identifier_expected.phone_number
+        assert phone_number_identifier_actual.identifier == phone_number_identifier_expected.identifier
 
     def test_serialize_teams_user(self):
         teams_user_identifier_model = CommunicationUserIdentifierSerializer.serialize(
-            MicrosoftTeamsUserIdentifier("teamsid")
+            MicrosoftTeamsUserIdentifier(
+                "teamsid",
+                cloud=CommunicationCloudEnvironment.Public,
+                identifier="someid"
+            )
         )
 
         assert teams_user_identifier_model.kind is CommunicationIdentifierKind.MicrosoftTeamsUser
-        assert teams_user_identifier_model.id is "teamsid"
+        assert teams_user_identifier_model.microsoft_teams_user_id is "teamsid"
+        assert teams_user_identifier_model.communication_cloud_environment is CommunicationCloudEnvironment.Public
+        assert teams_user_identifier_model.id is "someid"
 
     def test_deserialize_teams_user(self):
         teams_user_identifier_actual = CommunicationUserIdentifierSerializer.deserialize(
             CommunicationIdentifierModel(
+                id="someid",
                 kind=CommunicationIdentifierKind.MicrosoftTeamsUser,
                 microsoft_teams_user_id="teamsid",
-                is_anonymous=True
+                is_anonymous=True,
+                communication_cloud_environment="PUBLIC"
             )
         )
 
-        teams_user_identifier_expected = MicrosoftTeamsUserIdentifier("teamsid", is_anonymous=True)
+        teams_user_identifier_expected = MicrosoftTeamsUserIdentifier(
+            "teamsid",
+            identifier="someid",
+            cloud=CommunicationCloudEnvironment.Public,
+            is_anonymous=True
+        )
 
         assert isinstance(teams_user_identifier_actual, MicrosoftTeamsUserIdentifier)
+        assert teams_user_identifier_actual.identifier == teams_user_identifier_expected.identifier
         assert teams_user_identifier_actual.user_id == teams_user_identifier_expected.user_id
+        assert teams_user_identifier_actual.is_anonymous== teams_user_identifier_expected.is_anonymous
+        assert teams_user_identifier_actual.cloud == teams_user_identifier_expected.cloud
        
```

Comment:
Can we add two more tests?

1) Serialize an object that is foreign should throw
2) Deserialize something with a foreign value for kind should return an UnknownIdentifier